### PR TITLE
GO-4179 initial page web-publishing

### PR DESCRIPTION
**Pull request by zarkone**

### CODE REVIEW

File: core/publish/service.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
+package publish
```
Comment (deff7) [inline core/publish/service.go:1]: maybe `publisher`?

File: core/publish/service.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+func (s *service) Close(_ context.Context) error {
+	return nil
+}
+
+func (s *service) Name() (name string) {
+	return CName
+}
+
+func (s *service) publishById(ctx context.Context, spaceId, pageObjId string) (res PublishResult, err error) {
```
Comment (deff7) [inline core/publish/service.go:63]: unused

File: core/publish.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	"context"
+
+	"github.com/anyproto/anytype-heart/core/publish"
+	"github.com/anyproto/anytype-heart/pb"
+)
+
+func (mw *Middleware) ObjectPublish(ctx context.Context, req *pb.RpcObjectPublishRequest) *pb.RpcObjectPublishResponse {
+	publishService := getService[publish.Service](mw)
+
+	res, _ := publishService.Publish(ctx, req.SpaceId, req.ObjectId)
```
Comment (deff7) [inline core/publish.go:13]: check and handle error

File: core/publish/service.go (hunk 2 above)
Comment (zarkone) [reply core/publish/service.go]: right, this one supposed to use id but not string from client. Removed for now in b5105a60a9e40fcce775c9f9ed20772919df115c

**Comment by zarkone** [comment]
addressed comments and closing this PR in favor of https://github.com/anyproto/anytype-heart/pull/1684 -- will serve as an epic for feature

**Pull request closed without merging**